### PR TITLE
fix: use dedicated exception for Location validation

### LINKED ISSUE #11302: Relative path cannot start with /
**Describe the bug**
Since Vaadin Flow 14.6.1/2 the following warning is filling up our logs and eventually causing out-of-memory. It seems to be caused by a brute force attack targeting WordPress related files.

**To Reproduce**
Unknown

**Expected behavior**
Ideally Vaadin platform would detect brute-force attacks like this automatically and block it. If not, then please change the exception handling for incorrect relative paths such that it is not reported as a warning.

**Screenshots**
[qtp601893033-137] WARN com.vaadin.flow.router.Router - Exception when parsing location path /wp-includes/wlwmanifest.xml
java.lang.IllegalArgumentException: Relative path cannot start with /
	at com.vaadin.flow.router.Location.verifyRelativePath(Location.java:313)
	at com.vaadin.flow.router.Location.parsePath(Location.java:275)
	at com.vaadin.flow.router.Location.<init>(Location.java:74)
	at com.vaadin.flow.router.Router.getLocationForRequest(Router.java:115)
	at com.vaadin.flow.router.Router.resolveNavigationTarget(Router.java:153)
	at com.vaadin.flow.server.BootstrapUtils.resolvePageConfigurationHolder(BootstrapUtils.java:456)
	at com.vaadin.flow.server.BootstrapHandler$BootstrapContext.<init>(BootstrapHandler.java:208)
	at com.vaadin.flow.server.BootstrapHandler.createBootstrapContext(BootstrapHandler.java:1580)
	at com.vaadin.flow.server.BootstrapHandler.createAndInitUI(BootstrapHandler.java:1530)
	at com.vaadin.flow.server.BootstrapHandler.synchronizedHandleRequest(BootstrapHandler.java:474)
	at com.vaadin.flow.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40)
	at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1570)
	at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:252)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:750)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:755)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1617)
	at org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:226)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1604)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:545)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:590)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1607)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1297)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:485)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1577)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1212)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:221)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:146)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:500)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:547)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:270)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onFillable(SslConnection.java:543)
	at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:398)
	at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:161)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:388)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
	at java.base/java.lang.Thread.run(Thread.java:829)

### PR DESCRIPTION
**Pull request by denis-anisimov**
fixes #11302

## Description

Please list all relevant dependencies in this section and provide summary of the change, motivation and context.

Fixes # (issue)

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [ ] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [ ] I have added a description following the guideline.
- [ ] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

**Comment by pleku** [comment]
@denis-anisimov this adds what I consider public API (the bot added the `+0.1.0` label) and thus cannot be backport to 2.6 and 7.0.

For 8.0 it should be fine since we're not yet in beta.

File: flow-server/src/main/java/com/vaadin/flow/router/InvalidLocationException.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+     * 
+     * @param message
+     *            the detail message (which is saved for later retrieval by the
+     *            {@link #getMessage()} method).
+     * @param cause
+     *            the cause (which is saved for later retrieval by the
+     *            {@link #getCause()} method). (A <tt>null</tt> value is
+     *            permitted, and indicates that the cause is nonexistent or
+     *            unknown.)
+     * @since 1.4
```
Comment (pleku) [inline flow-server/src/main/java/com/vaadin/flow/router/InvalidLocationException.java:51]: Where does this come from ?

**Review by pleku — CHANGES_REQUESTED** [review]
Blocking for now since I've been working on #9443 and I'm not sure if it makes sense to make this change or not https://github.com/vaadin/flow/issues/11302#issuecomment-872032301

File: flow-server/src/main/java/com/vaadin/flow/router/InvalidLocationException.java (hunk 1 above)
Comment (denis-anisimov) [reply flow-server/src/main/java/com/vaadin/flow/router/InvalidLocationException.java:51]: copy paste from the RuntimeException docs

**Review by pleku — DISMISSED** [review]
Decided to approve this and I'll use the introduced exception to validate the `Location` in `BoostrapHandler` to adjust behavior for #9443 to return 400 code

**Review by pleku — APPROVED** [review]

**Pull request merged**